[MIG][15.0] auth_totp: add migration 15

daonamutc1 opened the pull request:
[Task](https://viindoo.com/web#id=22835&action=409&active_id=177&model=project.task&view_type=form&cids=1&menu_id=289)
Video:
- Befor: https://drive.google.com/file/d/1gEEFYBLAvqHwdmm0pwGBQ2EVTsMk0DW-/view?usp=sharing
- After: https://drive.google.com/file/d/1NV4ZZhAXLYO8LhinTOVtBAdty_NqpSR_/view?usp=sharing

royle-vietnam commented on openupgrade_scripts/scripts/auth_totp/15.0.1.0/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,18 @@
+
+---Fields in module 'auth_totp'---
+auth_totp    / res.users                / totp_trusted_device_ids (one2many): relation is now 'auth_totp.device' ('res.users.apikeys') [nothing to do]
+# NOTHING TO DO
```
@daonamutc1 sao nothing to do lại có post-migration thế anh?
Runbot lỗi pre commit anh nhé

royle-vietnam commented on openupgrade_scripts/scripts/auth_totp/15.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,17 @@
+    openupgrade.logged_query(
+        env.cr,
+        """INSERT INTO auth_totp_device (id, name, user_id, scope, index, key)
+            SELECT id, name, user_id, '%s', index, key 
```
@daonamutc1 thừa dấu cách ở cuối, tiện anh rebasse luôn nhé
```suggestion
            SELECT id, name, user_id, '%s', index, key
```

royle-vietnam commented on openupgrade_scripts/scripts/auth_totp/15.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,17 @@
+
+@openupgrade.migrate()
+def migrate(env, version):
+    # This method will copy records has scope = '2fa_trusted_device from res_users_apikeys to auth_totp_device
```
```suggestion
    # This method will copy records has scope = '2fa_trusted_device'
    # from res_users_apikeys to auth_totp_device
```

royle-vietnam commented on openupgrade_scripts/scripts/auth_totp/15.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
```suggestion
```

daonamutc1 committed:
[MIG] auth_totp: add migration for 15
- Copy records has scope = 2fa_trusted_device' from res_users_apikeys to auth_totp_device

royle-vietnam commented on openupgrade_scripts/scripts/auth_totp/15.0.1.0/post-migration.py:
```diff
@@ -0,0 +1,15 @@
+    # from res_users_apikeys to auth_totp_device
+    openupgrade.logged_query(
+        env.cr,
+        """INSERT INTO auth_totp_device (id, name, user_id, scope, index, key)
+            SELECT id, name, user_id, '%s', index, key
+            FROM res_users_apikeys
+            WHERE scope = '2fa_trusted_device'"""
+        % ("browser"),
```
@daonamutc1 như này có được không anh?
```suggestion
        """INSERT INTO auth_totp_device (id, name, user_id, scope, index, key)
            SELECT id, name, user_id, 'browser', index, key
            FROM res_users_apikeys
            WHERE scope = '2fa_trusted_device'"""
```

royle-vietnam replied:
@daonamutc1 đồng thời, xóa luôn các bản ghi có `scope = '2fa_trusted_device'` bên `res_users_apikeys` đi chứ nhỉ?

daonamutc1 committed:
[MIG] auth_totp: delete record from res_users_apikeys
